tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -182,26 +183,40 @@ def test_brand_string(test_microvm_with_api, network_config):
     "0x175",       # MSR_IA32_SYSENTER_ESP
     "0x176",       # MSR_IA32_SYSENTER_EIP
     "0x6e0",       # MSR_IA32_TSCDEADLINE
+    "0x834"     ,  # LVT Performance Monitor Interrupt Register.
     "0xc0000082",  # MSR_LSTAR
     "0xc0000083",  # MSR_CSTAR
     "0xc0000100",  # MSR_FS_BASE
     "0xc0000101",  # MSR_GS_BASE
+    "0xc0010007",  # MSR_K7_PERFCTR3
+    "0xc001020b",  # Performance Event Counter MSR_F15H_PERF_CTR5
+    "0xc0011029",  # MSR_F10H_DECFG also referred to as MSR_AMD64_DE_CFG
 ]
+
+# CPU templates which support MSR test
+SUPPORTED_CPU_TEMPL = ["T2S", "T2A"]
+SUPPORTED_CPU_VENDORS = [
+                            cpuid_utils.CpuVendor.INTEL,
+                            cpuid_utils.CpuVendor.AMD
+                        ]
 # fmt: on
 
 
 @pytest.mark.skipif(
     PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
 )
 @pytest.mark.skipif(
-    cpuid_utils.get_cpu_vendor() != cpuid_utils.CpuVendor.INTEL,
-    reason="CPU templates are only available on Intel.",
+    cpuid_utils.get_cpu_vendor() not in SUPPORTED_CPU_VENDORS,
+    reason=f"CPU templates are only available on \
+                {[v.name for v in SUPPORTED_CPU_VENDORS]}",
 )
 @pytest.mark.skipif(
     utils.get_kernel_version(level=1) not in SUPPORTED_KERNELS,
     reason=f"Supported kernels are {SUPPORTED_KERNELS}",
 )
-@pytest.mark.parametrize("cpu_template", ["T2S"])
+@pytest.mark.parametrize(
+    "cpu_template", set(SUPPORTED_CPU_TEMPLATES).intersection(SUPPORTED_CPU_TEMPL)
```

Comment:
so this can now be reworked similar to https://github.com/firecracker-microvm/firecracker/blob/main/tests/integration_tests/functional/test_feat_parity.py#L114 and others in that file.